### ISSUE TITLE
Handling metadata on special files 404/500 pages

### ISSUE THREAD

**Issue by SBAVM** [marko-js/run#28]
I use the +layout.marko file for all pages with routes. Given that the +404.marko and +500.marko files are living as standalone files in the root of the folder I'm not able to pull any metadata into these pages as I can with normal routes.

My +layout.marko looks like this
``` html
<!DOCTYPE html>
<html dir='ltr' lang=($global.meta.pageLanguage)>
  <head>
    <meta charset='UTF-8'>
    ...
    <title>${$global.meta.pageTitle}</title>
  </head>
  <body>
    <${input.renderBody}/>
  </body>
</html>
```

Is there something I oversee?
From my point of view it would be nice to see the 404 and 500 pages live in special folders named `_404` and `_500` where files like `+page.marko` and `+meta.*` find a place.

**Comment by rturnq** [comment]
Currently the `+500.marko` error page will have access to the context object for the route which was executing when the error occurred. This means it will have access to the route information for that route including the meta data. I think that makes sense as you might want to use that route information when rendering the error for the user. The 404 page however doesn't have access to any meta data as it doesn't have a "route" associated with it.

We are leaning towards associating the adjacent `+meta.*` file with the `+404.marko` page which would be the root meta file and then also allowing the `+404.marko` file to be defined within a pathless route which would let you isolate it's meta file.

So basically you could have something like
```
/routes
  /_404
    +404.marko
    +meta.json
  +page.marko
```
And this would be picked up as the 404 page and be associated with its own meta file. The folder name `_404` could be anything as long as it started with '_' to make it pathless.

I am a bit torn on this though. I like your suggestion of using the folder name to signify the special routes. This would let us be consistent with the files and semantics; however, I'm not totally sure the 404 and 500 pages should be full fledged routes with their own handler or middleware.
